CMakeLists.txt

```diff
@@ -6,120 +6,8 @@ set(VERSION 1.0.5)
 
 set(CMAKE_C_STANDARD 99)
 
-add_subdirectory(zlib)
-
-add_library(pplib_utils STATIC
-    util/utilarm.h
-    util/utilplat.h
-    util/utilcryptdef.h
-    util/utildecl.h
-    #
-    util/utilcrypt.c
-    util/utilcrypt.h
-    util/utiliof.c
-    util/utiliof.h
-    util/utillog.c
-    util/utillog.h
-    util/utilmd5.c
-    util/utilmd5.h
-    util/utilmem.c
-    util/utilmem.h
-    util/utilnumber.c
-    util/utilnumber.h
-    util/utilsha.c
-    util/utilsha.h
-)
-
-target_include_directories(pplib_utils PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/util>
-)
-
-target_link_libraries(pplib_utils
-    PUBLIC
-        m
-        zlib
-)
-
-add_library(pplib_filters STATIC
-    util/utilbasexx.c
-    util/utilbasexx.h
-    util/utilflate.c
-    util/utilflate.h
-    util/utilfpred.c
-    util/utilfpred.h
-    util/utillzw.c
-    util/utillzw.h
-)
-
-target_include_directories(pplib_filters PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/util>
-)
-
-target_link_libraries(pplib_filters
-    PUBLIC
-        m
-        zlib
-)
-
-add_library(pplib
-    ppapi.h
-    ppconf.h
-    pplib.h
-    #
-    pparray.c
-    pparray.h
-    ppcrypt.c
-    ppcrypt.h
-    ppdict.c
-    ppdict.h
-    ppfilter.h
-    ppheap.c
-    ppheap.h
-    ppload.c
-    ppload.h
-    ppstream.c
-    ppstream.h
-    ppxref.c
-    ppxref.h
-)
-
-target_include_directories(pplib PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/util>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-)
-
-target_link_libraries(pplib
-    PRIVATE
-        pplib_filters
-        pplib_utils
-)
-
-target_include_directories(pplib
-    PRIVATE
-        util
-)
-set_target_properties(pplib
-    PROPERTIES
-        PUBLIC_HEADER "ppapi.h;ppconf.h;pplib.h"
-)
-
-add_library(pplib::pplib ALIAS pplib)
-
-install(TARGETS pplib
-    EXPORT pplib
-    DESTINATION lib
-)
+find_package(ZLIB REQUIRED)
```

Comment:
I don't want to spend way too much effort on this library in any case and I'm sorry if this was a misunderstanding.

What initially bothered me was that:
* In the long term it could be nice to offer an option to decide between external and internal zlib (low priority though), similar to what TeX Live offers.
* Even if I specified the installation directory, the build system complained that it could not install `zlib` to `/usr/local`. If internal zlib is used, I would not want to install it, to start with.

But please note that this is a super minor complaint. Our primary goal at the moment is to set up a testing infrastructure and fix some bugs on arm etc. So it's not all that important to have the complete flexibility in place. It's probably way more important to be able to easily build the library everywhere, and allowing just external zlib is probably more of a problem in that case?

Reply:
To have an architecture similar to TeX Live, we would probably want to place libraries in a separate directory and build them using [`ExternalProject_Add`](https://cmake.org/cmake/help/latest/module/ExternalProject.html), then adjust [`CMAKE_PREFIX_PATH`](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html) so that they can be found by `find_package`.  The whole thing could go in a `if()` block to switch it on and off on demand.

Reply:
From the very beginning, I link pplib with auxilary zlib, so this approach is ok to me.